### ISSUE TITLE
Set and Sequence in structured

### ISSUE THREAD

**Issue by bottler** [omry/omegaconf#968]
In version 2.1, `typing.Set` and `typing.Sequence` could be used in structured config. This no longer works in 2.2.2. The feature is not that important to me, but if this is not meant to work then the error should be made more friendly.

The error is
```
  File ".../site-packages/omegaconf/omegaconf.py", line 1028, in _node_wrap
    elif issubclass(ref_type, Enum):
omegaconf.errors.ConfigTypeError: issubclass() arg 1 must be a class
```
which is quite unfriendly (e.g. there is no indication where in my code there might be a problem).

**To Reproduce**
```python
from dataclasses import dataclass
from omegaconf import OmegaConf, DictConfig
from typing import Sequence, Set

@dataclass
class A:
    ii: Sequence[int] = (3, 43)

out = OmegaConf.structured(A)
```

or

```python
@dataclass
class A:
    ii: Set[int] = {3, 43}

out=OmegaConf.structured(A)
```

**Expected behavior**
In 2.1, `structured` worked in this case, and the actual member `out.ii` was a ListConfig. This was a convenience. In 2.2.2 I would expect this to either continue to work or to have an error message that `Sequence` (respectively `Set`) is not allowed.

**Additional context**
- [ ] OmegaConf version: 2.2.2
- [ ] Python version: 3.8
- [ ] Operating system: Linux

**Comment by Jasha10** [comment]
> `inspect.isclass(ref_type) and issubclass(ref_type, Enum)` - to avoid this errror from all manner of things.

Great idea -- thanks!

**Comment by Jasha10** [comment]
Closed by #993.
